### DISCUSSION TITLE
How can I commit into main repo.

### DISCUSSION THREAD

**Discussion by lav023** [OWASP/www-project-csrfguard#88] [Q&A] [answered]
Hi, 

There are few changes which my company wants to make into main repo, they believe that these few changes are going to help everyone in the community, can you please share me the details through which we can raise the Pull request .

**Comment by forgedhallpass** [comment, marked as answer]
Hello @lav023,

The workflow is the same as for any other projects:
1. fork the repository
2. clone your fork
3. implement your feature/bugfix
4. commit + push your changes
5. create a pull request on the parent repository

See: 
* https://docs.github.com/en/get-started/quickstart/fork-a-repo
* https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork
* https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/working-with-your-remote-repository-on-github-or-github-enterprise/creating-an-issue-or-pull-request

**NOTE**: Before creating PRs though, you should **create an [`Issue`](https://github.com/OWASP/www-project-csrfguard/issues/new/choose) first** explaining what you would actually like to achieve (e.g. adding or requesting a new feature, reporting a bug, optimizing performance etc.)
